src/adapter/adapter.ts

```diff
@@ -170,11 +180,11 @@ abstract class Adapter extends events.EventEmitter {
     public abstract addInstallCode(ieeeAddress: string, key: Buffer): Promise<void>;
 
     public abstract waitFor(
-        networkAddress: number,
+        networkAddress: number | undefined,
```

Comment:
@Nerivec why did you make these optional? I see it only being used in `endpoint.ts` where it is never `undefined`, same goes for the `transactionSequenceNumber`

Reply:
Internally it's used by adapters that sometimes can't pass it:
https://github.com/Koenkk/zigbee-herdsman/blob/16d5cfe058ccb20af0e0d48356b577a0d31ce64b/src/adapter/z-stack/adapter/zStackAdapter.ts#L1036-L1045